lib/perl/Genome/VariantReporting/Framework/Component/Report/MergeCompatible.pm

```diff
@@ -23,4 +23,10 @@ sub report_path {
         $self->class);
 }
 
+sub has_size {
+    my $self = shift;
+
+    return -s $self->report_path;
+}
```

Comment:
You could make it `-s $self->report_path || 0` so that it is `defined`.  (Or, if you prefer, something like `-s $self->report_path? 1 : 0` to normalize all values.)

Another form of memoization without using `Memoize` that has been used here is [this](https://github.com/genome/genome/blob/6d979af90a3558c93447477f46e950122aa21bc0/lib/perl/Genome/FeatureList/Command/Import.pm#L129-L137), where a particular hash key on `$self` is checked and set explicitly.